doc/devel/testing.rst

```diff
@@ -53,58 +58,64 @@ value.
 
-Running the tests is simple. Make sure you have nose installed and run::
+Running the tests is simple. Make sure you have pytest installed and run::
+
+   py.test
+
+or::
 
    python tests.py
 
 in the root directory of the distribution. The script takes a set of
 commands, such as:
 
 ========================  ===========
-``--pep8``                pep8 checks
-``--no-pep8``             Do not perform pep8 checks
-``--no-network``          Disable tests that require network access
+``--pep8``                Perform pep8 checks (requires pytest-pep8_)
+``-m "not network"``      Disable tests that require network access
 ========================  ===========
 
-Additional arguments are passed on to nosetests. See the nose
-documentation for supported arguments. Some of the more important ones are given
-here:
+Additional arguments are passed on to pytest. See the pytest documentation for
+`supported arguments`_. Some of the more important ones are given here:
 
 =============================  ===========
 ``--verbose``                  Be more verbose
-``--processes=NUM``            Run tests in parallel over NUM processes
-``--process-timeout=SECONDS``  Set timeout for results from test runner process
-``--nocapture``                Do not capture stdout
+``--n NUM``                    Run tests in parallel over NUM
+                               processes (requires pytest-xdist_)
+``--timeout=SECONDS``          Set timeout for results from each test
+                               process (requires pytest-timeout_)
+``--capture=no`` or ``-s``     Do not capture stdout
 =============================  ===========
 
-To run a single test from the command line, you can provide a
-dot-separated path to the module followed by the function separated by
-a colon, e.g., (this is assuming the test is installed)::
+To run a single test from the command line, you can provide a file path,
+optionally followed by the function separated by two colons, e.g., (tests do
+not need to be installed, but Matplotlib should be)::
 
-  python tests.py matplotlib.tests.test_simplification:test_clipping
+  py.test lib/matplotlib/tests/test_simplification.py::test_clipping
+
+or, if tests are installed, a dot-separated path to the module, optionally
+followed by the function separated by two colons, such as::
+
+  py.test --pyargs matplotlib.tests.test_simplification::test_clipping
 
 If you want to run the full test suite, but want to save wall time try
 running the tests in parallel::
 
-  python tests.py --nocapture --verbose --processes=5 --process-timeout=300
+  py.test --verbose -n 5
 
+Depending on your version of Python and pytest-xdist, you may need to set
+``PYTHONHASHSEED`` to a fixed value when running in parallel::
 
-An alternative implementation that does not look at command line
-arguments works from within Python is to run the tests from the
-matplotlib library function :func:`matplotlib.test`::
+  PYTHONHASHSEED=0 py.test --verbose -n 5
```

Comment:
Can we just specify a minimum version of pytest-xdist where this is no longer an issue?  It looks like it is related to https://bitbucket.org/pytest-dev/pytest/issues/346/pytest-xdist-and-python-33-is-sort-of, which has been resolved in 2013.